### PR TITLE
daemon: validate runtime.usage for canonicalisation-safety

### LINKED ISSUE #1008: daemon: validate emitter-supplied open containers before embedding in the signed receipt
`EmitterTool.Usage` (`daemon/internal/pipeline/build.go:150`) is a `json.RawMessage` forwarded **verbatim** into `issuer.runtime.usage` (`build.go:935`), per the ADR-0026 open-container design. The hook supplies it by reading `message.usage` straight out of the Claude Code transcript file (`hook/cmd/obsigna-hook/claude_transcript.go:23-25`, `:108`).

So arbitrary JSON from an external artifact is embedded inside the **signed, canonicalised** receipt document with no well-formedness validation.

Being *open* is deliberate and correct. Being *unvalidated for canonicalisation-safety* is the gap: it means the document canonicaliser cannot assume its input is well-formed, which constrains how strict that canonicaliser is allowed to be. Any move toward stricter canonicalisation would risk a malformed `usage` blob failing the whole receipt build — punching exactly the audit-trail hole that `build.go:836-838` argues against.

Size is bounded (the 1 MiB frame cap covers the whole frame).

### Suggested fix
Validate emitter-supplied open containers at the trust boundary and, on violation, **drop the field with a logged warning while still emitting the receipt** — mirroring the existing disclosure fallback pattern (encryption failure degrades to hash-only, never drops the event).

_Surfaced by a code-review pass._

### PR DESCRIPTION
**Pull request by ojongerius**
## What

`issuer.runtime.usage` (`daemon/internal/pipeline/build.go`) is now validated for RFC 8785 canonicalisation-safety at the trust boundary before it is embedded in the signed receipt. On a violation, the field is dropped and a warning is logged via `Pipeline.ErrorLog` — the receipt is still built, signed, and stored.

`issuerFromFrame` became a `Pipeline` method (was a free function taking `daemonID string`) so it can reach `p.logError`. A new `validatedUsage` helper attempts `receipt.Canonicalize` on the raw `usage` payload and reports whether it's safe to embed.

## Why

`EmitterTool.Usage` is a `json.RawMessage` forwarded **verbatim** into `issuer.runtime.usage`, per the ADR-0026 open-container design — arbitrary JSON from an external artifact (the Claude Code transcript, for the in-tree hook) with no well-formedness validation.

Being *open* is deliberate. Being *unvalidated for canonicalisation-safety* was the gap: the field reached the signer unchecked, so a syntactically-valid-but-uncanonicalisable payload — e.g. a JSON number like `1e400`, which overflows `float64` when `receipt.Canonicalize` re-parses it — would fail canonicalisation of the *whole* receipt during signing, and `Process` would return an error and drop the entire event. That's exactly the audit-trail hole the pipeline's existing comments (`build.go`) argue against for `Input`/`Output` hashing failures.

The fix mirrors the existing disclosure fallback pattern already used for forensic encryption failures: degrade (drop the field, log a warning), never drop the event.

Closes #1008.

## Testing

- `go vet ./...` and `go test ./...` in `daemon/` — all pass.
- Added `TestProcess_DropsUncanonicalizableUsage`, mirroring the existing `TestProcess_RejectsUnrepresentableNumbers` fixture (`1e400`) but for `usage` instead of `input`/`output`: asserts `Process` succeeds, exactly one warning is logged, `runtime.usage` is absent from the stored receipt while `runtime.model`/`capture_method` survive, and the receipt still re-hashes cleanly.

## Checklist

- [x] Tests pass for all changed components (`daemon`)
- [x] Linter passes (`go vet`)
- [x] No real keys or secrets in the diff
- [ ] Cross-language tests pass (if receipt format, signing, or hashing changed) — N/A, no receipt schema or wire-format change
- [ ] AGENTS.md updated (if project structure changed) — N/A
- [ ] Spec changes have been reviewed by a maintainer (if applicable) — N/A, no spec change

## Security

- [x] This PR touches crypto, auth, or secrets handling — it touches what gets embedded in a *signed* receipt (not crypto primitives themselves)
- [x] Primitives and parameters have been reviewed — no crypto primitives changed
- [x] All inputs are validated at trust boundaries — `runtime.usage` is now validated before being embedded in signed output
- [x] Edge cases are tested (nil, empty, corrupted, concurrent) — covered by the new test plus pre-existing `TestProcess_RuntimeModelUsageOnRootReceipt` (valid usage) and `TestProcess_RejectsUnrepresentableNumbers` (the analogous input/output case)

### CODE REVIEW

File: daemon/internal/pipeline/build.go [hunk 1]
```diff
@@ -944,6 +944,32 @@ func issuerFromFrame(f *EmitterFrame, daemonID string) receipt.Issuer {
...
+// whatever they report), so the daemon cannot assume it is well-formed enough
+// to survive RFC 8785 canonicalisation — e.g. a number like 1e400 is
+// syntactically valid JSON but overflows float64 when Canonicalize re-parses
+// it (see canonicalSHA256). Unlike Input/Output, Usage is not required for
+// the receipt to be a valid audit record, so a violation here MUST NOT fail
+// the whole receipt build the way a bad Input/Output hash does. Instead it is
+// dropped with a logged warning — mirroring the disclosure-encryption
+// fallback (degrade, never drop the event) — so the receipt still gets
+// signed and stored via the normal path, and the failure surfaces later, once
+// only, when the whole receipt is canonicalised for hashing/signing.
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/internal/pipeline/build.go]: Good catch — the comment was wrong, `validatedUsage` detects and logs the failure eagerly, not later during whole-receipt canonicalisation. Fixed in 4bce296.
